pages/api/auth/login.js

```diff
@@ -0,0 +1,34 @@
+import Cookies from 'cookies';
+import User from '../../../models/User';
+import CustomError from '../../../helpers/api/CustomError';
+import catchErrors from '../../../helpers/api/catchErrors';
+
+const handler = async (req, res) => {
+	if (req.method === 'POST') {
+		const user = await User.findOne({ email: req.body.email });
+
+		if (!user) {
+			throw new CustomError('Invalid credentials', 401);
```

Comment:
We could do better with errors.
1. Give more info. Here we can return "Email is not registered". If the user exists but the password is incorrect, the app should say Incorrect password.
2. More structured errors. We are showing error text at the bottom of the form but many times errors are related to specific fields rather than the form as a whole. We could have more structured errors in the form of an array 
e.g.
```js
[
    {
        field: "email",
        error: "Email is not registered"
    }
]
```
The UI would have to then parse this and display the errors at each field.

Reply:
Why don't we just send { email: "Email is not registered" } ?

Reply:
Thats ok too as long as we only have one error per field.

Reply:
I Think That will be the case most of the time in this app.

> Thats ok too as long as we only have one error per field.